var/spack/repos/builtin/packages/gcc-runtime/package.py

```diff
@@ -0,0 +1,151 @@
+# Copyright 2013-2023 Lawrence Livermore National Security, LLC and other
+# Spack Project Developers. See the top-level COPYRIGHT file for details.
+#
+# SPDX-License-Identifier: (Apache-2.0 OR MIT)
+
+import os
+import re
+
+from macholib import MachO, mach_o
+
+from spack.package import *
+from spack.util.elf import parse_elf
+
+
+class GccRuntime(Package):
+    """Package for GCC compiler runtime libraries"""
+
+    homepage = "https://gcc.gnu.org"
+    has_code = False
+
+    tags = ["runtime"]
+
+    maintainers("haampie")
+
+    license("GPL-3.0-only")
+
+    requires("%gcc")
+
+    LIBRARIES = [
+        "asan",
+        "atomic",
+        "gcc_s",
+        "gfortran",
+        "gomp",
+        "hwasan",
+        "itm",
+        "lsan",
+        "quadmath",
+        "ssp",
+        "stdc++",
+        "tsan",
+        "ubsan",
+    ]
+
+    def install(self, spec, prefix):
+        if spec.platform in ["linux", "cray", "freebsd"]:
+            libraries = self._get_libraries_elf()
+        elif spec.platform == "darwin":
+            libraries = self._get_libraries_macho()
+        else:
+            raise RuntimeError("Unsupported platform")
+
+        if not libraries:
+            raise RuntimeError("Failed to detect GCC runtime libraries")
+
+        mkdir(prefix.lib)
+
+        for path, name in libraries:
+            install(path, os.path.join(prefix.lib, name))
+
+    def _get_libraries_elf(self):
+        """Get the GCC runtime libraries for ELF binaries"""
+        cc = Executable(self.compiler.cc)
+        lib_regex = re.compile(rb"\blib[a-z-_]+\.so\.\d+\b")
+        path_and_install_name = []
+
+        for name in self.LIBRARIES:
+            # Look for the dynamic library that gcc would use to link,
+            # that is with .so extension and without abi suffix.
+            path = cc(f"-print-file-name=lib{name}.so", output=str).strip()
+
+            # gcc reports an absolute path on success
+            if not os.path.isabs(path):
+                continue
```

Comment:
Would make spack unusable when gcc is configured with `--disable-shared`

Reply:
I've pushed a commit that warns when nothing is found. It's certainly wrong to error.

Reply:
If it’s wrong to error with disable-shared, maybe leaving it is fine. Warning is probably good for now… ideally we could tell if it’s supposed to have —disable-shared and only fail when we expect no runtime libs.

With disable-shared there’s not really a dependency on the runtime libs anyway, or at least not the same sort of dep, so that’s yet another modeling problem to deal with.

Reply:
Yeah, unfortunately that info is lost, so `gcc-runtime` in a build cache is no guarantee it'll work with other people's gcc at the same version : (. We could deal with it later: `gcc-runtime +shared-X` or something.

Reply:
We can also split libraries according to e.g. languages later